mapr/datadog_checks/mapr/mapr.py

```diff
@@ -0,0 +1,153 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import json
+import os
+import re
+
+from six import iteritems
+
+from datadog_checks.base import AgentCheck, ensure_unicode
+from datadog_checks.base.errors import CheckException
+
+from .common import ALLOWED_METRICS, COUNT_METRICS, GAUGE_METRICS, MONOTONIC_COUNTER_METRICS
+from .utils import get_fqdn, get_stream_id_for_topic
+
+try:
+    # The `confluent_kafka` library here is the one made by mapr
+    import confluent_kafka as ck
+except ImportError:
+    ck = None
+
+
+DEFAULT_STREAM_PATH = "/var/mapr/mapr.monitoring/metricstreams"
+METRICS_SUBMITTED_METRIC_NAME = "mapr.metrics.submitted"
+SERVICE_CHECK = "mapr.can_connect"
+
+"""
+All mapr metrics go through a Stream topic (similar to a Kafka topic) and are consumed by OpenTSDB, a timeserie
+database.
+All the metrics are distributed over multiple topics, one for each host. This way, the check instance can subscribe
+to the topic relevant to the current host and consume everything.
+
+Note: The MapR documentation https://mapr.com/docs/61/AdministratorGuide/spyglass-on-streams.html states that
+there is one topic per host per metric name. This is wrong starting with 6.1+, there is only one topic per host now.
+To support older versions of MapR, the check should be updated to subscribe those multiple topics.
+"""
+
+
+class MaprCheck(AgentCheck):
+    def __init__(self, name, init_config, instances):
+        super(MaprCheck, self).__init__(name, init_config, instances)
+        self._conn = None
+        self.hostname = self.instance.get('hostname', get_fqdn())
+        self.streams_count = self.instance.get('streams_count', 2)
+        self.topic_path = "{stream_path}/{stream_id}:{topic_name}".format(
+            stream_path=self.instance.get('stream_path', DEFAULT_STREAM_PATH),
+            stream_id=get_stream_id_for_topic(self.hostname, rng=self.streams_count),
+            topic_name=self.hostname,
+        )
+        self.allowed_metrics = [re.compile(w) for w in self.instance.get('metric_whitelist', [])]
+        self.base_tags = self.instance.get('tags', [])
+        self.is_first_check_run = True
+
+        auth_ticket = self.instance.get('ticket_location')
+        if auth_ticket:
+            os.environ['MAPR_TICKETFILE_LOCATION'] = auth_ticket
+        elif not os.environ.get('MAPR_TICKETFILE_LOCATION'):
+            self.log.info("MAPR_TICKETFILE_LOCATION environment variable not set, this may cause authentication issues")
```

Comment:
I first put it at an info level because some users may not have this security on their clusters. But from what I understand now this is a really standard usage, I'll change it to warn